cypress-io/cypress
# chore: migrate runner/dom.js

- Part 2 of #21689

### User facing changelog

N/A. It's an internal refactoring.

### Additional details

- Why was this change necessary? => Some legacy codes are left inside `runner`. This PR migrates `dom` codes. 
- What is affected by this change? => N/A

### Any implementation details to explain?

To migrate `runner/dom` to `pkg/app`, these things had to be done:

  * Migrate runner/dom.js
  * Mgirate Highlight React component to Vue. 
  * Migrate AssertionsMenu React component to Vue. 
  * Remove JQuery as much as possible. 

The result is like this:

* `src/runner/dom.ts` for common codes for `Highlight` and `AssertionsMenu`.
* If it's related to placing `Highlight` correctly, I moved it inside `aut-iframe`.
* I moved `Highlight` to `runner/selector-playground`.
* I moved `AssertionsMenu` to `runner/studio`. And I splitted the file to 3 component files. 

I used `floating-ui` rather than our `Tooltip` because `floating-vue` doesn't support shadow dom. It seems that it's fixed after floating-ui@1.x.

### Steps to test

* Check if Selector Playground's `Highlight` works correctly.
* Check if Studio's Assertions menu works correctly.

### How has the user experience changed?

N/A.

### PR Tasks


- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
